l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py

```diff
@@ -0,0 +1,320 @@
+        ''' Go back to the finish view.
+            Called by pressing back button.
+
+            :returns action: configuration for the next wizard's view
+        '''
+        self.ensure_one()
+        self._state_finish_on()
+        return self._do_populate_tasks()
+
+    @api.multi
+    def cancel_button(self):
+        ''' Remove the attachment create.
+            Called by pressing cancel button.
+
+            :returns: None
+        '''
+        self.ensure_one()
+        self.attachment_id.unlink()
+
+    ##############################
+    #          function          #
+    ##############################
+    @api.multi
+    def _get_sftp_key(self):
+        ''' private function that get the sftp key need for connection
+            with the server.
+
+            :returns (str key, str key_pass):
+            :riases Warning: if no key found
+        '''
+        fds_authentication_key_obj = self.env['fds.authentication.keys']
+        key = fds_authentication_key_obj.search([
+            ['user_id', '=', self.env.uid],
+            ['fds_account_id', '=', self.fds_account.id]])
+
+        if not key:
+            raise exceptions.Warning("You don't have a key")
+
+        if not key.key_active:
+            raise exceptions.Warning('Key not active')
+
+        key_pass = fds_authentication_key_obj.config()
+        return (key, key_pass)
+
+    @api.multi
+    def _create_tmp_file(self, data, tmp_directory=None):
+        ''' private function that write data to a tmp file and if no tmp
+            directory use, create one.
+
+            :param str data: data in base64 format
+            :param str tmp_directory: path of the directory
+            :returns (obj file, str directory): obj of type tempfile
+        '''
+        self.ensure_one()
+        try:
+            if not tmp_directory:
+                tmp_directory = tempfile.mkdtemp()
+
+            tmp_file = tempfile.NamedTemporaryFile(dir=tmp_directory)
+            tmp_file.write(base64.b64decode(data))
+            tmp_file.flush()
+            return (tmp_file, tmp_directory)
+        except Exception as e:
+            _logger.error("Bad handling tmp in fds_inherit_sepa_wizard: %s", e)
+
+    @api.multi
+    def _add2historical(self):
+        ''' private function that add the upload file to historic
+
+            :returns: None
+        '''
+        self.ensure_one()
+        values = {
+            'payment_order_id': self.payment_order_id.id,
+            'fds_account_id': self.fds_account.id,
+            'filename': self.filename,
+            'directory_id': self.fds_account_directory.id,
+            'state': 'uploaded'}
+        historical_dd_obj = self.env['fds.postfinance.historical.sepa']
+        historical_dd_obj.create(values)
+
+    @api.multi
+    def _state_finish_on(self):
+        ''' private function that change state to finish
+
+            :returns: None
+        '''
+        self.ensure_one()
+        self.write({'state': 'finish'})

```

Comment:
As you ensure you have one record, you can write `self.state = `finish` otherwise you can remove `self.ensure_one()`